### ISSUE TITLE
CrossValidation criterion ignored, train metric used instead

### ISSUE THREAD

**Issue by borguz** [Microsoft/CNTK#2571]
I'm using training_session to train a model.  I pass a training loss and metric to Trainer as follows:
`Trainer(None, (model.train_loss, model.metric), learner)`
Then I define a cross validation configuration:
`cv_cfg = CrossValidationConfig(minibatch_source=valid_src, frequency=epoch_size,
                                       minibatch_size=batch_size,
                                       criterion=model.eval_loss)`
Then I run training:
`training_session(trainer, train_src, mb_schedule, input_map, max_samples=samples, 
                         progress_frequency=log_steps, 
                         checkpoint_config=ckpt_cfg, cv_config=cv_cfg).train()`

When cross validation happens, model.metric is evaluated, where I would expect model.eval_loss that is passed to CrossValidationConfig be used instead.

**Comment by jsundh** [comment]
What is the status on this? I found myself wanting a separate CV criterion for a project as well.

Since `TrainingSession::CrossValidate` [only calls](https://github.com/Microsoft/CNTK/blob/master/Source/CNTKv2LibraryDll/TrainingSession.cpp#L299) `Trainer::TestMinibatch`, it seems more like a missing feature than a bug, but I think it would be a nice and usable feature to have. A not at all far-fetched example would be sampled softmax for training and full softmax for cross validation.
